lib/from_honeybee/_openapi/HoneybeeMapper.rb

```diff
@@ -0,0 +1,127 @@
+# Energy, LLC, Ladybug Tools LLC and other contributors. All rights reserved.
+# 
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# (1) Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# (2) Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# (3) Neither the name of the copyright holder nor the names of any contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'urbanopt/scenario'
+require 'openstudio/common_measures'
+require 'openstudio/model_articulation'
+
+require 'json'
+
+module URBANopt
+    module Scenario
+      class HoneybeeMapper < SimulationMapperBase
+
+      # class level variables
+      @@instance_lock = Mutex.new
+      @@osw = nil
+      @@geometry = nil
+
+      def initialize()
+
+        # do initialization of class variables in thread safe way
+        @@instance_lock.synchronize do
+            if @@osw.nil?
+  
+              # load the OSW for this class
+              osw_path = File.join(File.dirname(__FILE__), 'honeybee_workflow.osw')
+              File.open(osw_path, 'r') do |file|
+                @@osw = JSON.parse(file.read, symbolize_names: true)
+              end
+
+            # configures OSW with extension gem paths for measures and files, all extension gems must be
+            # required before this
+            @@osw = OpenStudio::Extension.configure_osw(@@osw)
+            end
+          end
+      end
+
+      def create_osw(scenario, features, feature_names)
+
+        if features.size != 1
+          raise "TestMapper1 currently cannot simulate more than one feature"
```

Comment:
What is the reason for this limitation? Is the only thing that we need to do to get rid of it is put all the code below this into a `for each` loop?

Reply:
Currently the Mapper is creating an OSW file for each feature. The `create_osw` method is passing an array which consists of only 1 feature and is therefore getting the first element from the array.

Reply:
Ah, I think I see now. So the ability to pass multiple features is just for some future workflow where you might want to generate a single osw for multiple features (like if all of the buildings need to go into a single OpenStudio model). Ok. This is good, then.